www/matj27/hw08/homework.js

```diff
@@ -0,0 +1,321 @@
+/* HOMEWORK */
+/**
+ * 0) Pre-preparacion.
+ * - Vytvořte HTML stránku s nadpisem h1 "JavaScript is awesome!"
+ * - Na stránce vytvořte místo pro umístění jednotlivých spouštěčů úkolů - tlačítek (tj. div, který má id s hodnotou
+ *   "task-buttons").
+ * - Na stránce vytvořte místo pro výpis výsledků úkolů (div, který má id s hodnotou "result").
+ *
+ * - Připojte tento homework.js soubor k vytvořené HTML stránce pomocí tagu <script> (viz LAB) a vyzkoušejte
+ * console.log('Ahoj světe');
+ */
+console.log('Ahoj světe');
+
+
+/**
+ * 1) Pepe's age. Vypište na konzoli smysluplnou oznamovací větu s věkem Pepy, pokud znáte jeho rok narození,
+ * který je uložený v proměnné a pro výpis použijte zřetězení stringů nebo interpolaci. Pro názvy proměnných
+ * používejte smysluplnou angličtinu.
+ */
+// Solution here
+const stringPepesAge = (age) => {
+    return `Pepe is ${age} years old.`;
+};
+
+console.log(stringPepesAge(20));
+
+/**
+ * 2) WTF (wow, that's fun). Vypište na konzoli teplotu v Fahrenheiht, pokud znáte teplotu v Celsius, a také naopak.
+ * Formát výpisu je: 20°C =  68°F resp. 68°F = 20°C. Opět používejte proměnné. Výpočet probíhá takto:
+ *     z C na F: vynásobit 9, vydělit 5 a přičíst 32.
+ *     z F na C: odečíst 32, vynásobit 5 a vydělit 9.
+ */
+// Solution here
+const celsiusToFahrenheiht = (celsius) => {
+    return `${celsius}°C = ${(celsius * 9 / 5) + 32}°F`;
+};
+
+const fahrenheihtToCelsius = (fahrhenheiht) => {
+    return `${fahrhenheiht}°F = ${(fahrhenheiht - 32) * 5 / 9}°C`;
+};
+
+const celsius = 20;
+console.log(celsiusToFahrenheiht(celsius));
+
+const fahrhenheiht = 68;
+console.log(fahrenheihtToCelsius(fahrhenheiht));
+
+/**
+ * 3) Funkce function fonction funktio. Vemte předchozí úlohy a udělejte z nich funkce. Tj. vytvoříte funkce,
+ * které přijímají argumenty, a na základě argumentů po zavolání vypíše výsledek na konzoli.
+ * Párkrát zavolejte tyto funkce s různými argumenty. V konzoli také vyzkoušejte, zda fungují vaše funkce.
+ *
+ * Pro testování funkce:
+ * - Pouze pomocí JavaScriptu (bez knihoven) vytvořte HTML tlačítko s názvem této úlohy, resp. co funkce dělá, a
+ * id s číslem úlohy <button id="task-1">Uloha 1 (Pepe's age)</button>, umístěte ho na stránku do předem vytvořeného
+ * místa <div id="tasks"></div> a pomocí posluchače události "click" nabindujte implementovanou funkci na toto tlačítko.
+ *
+ * Výsledkem má být tlačítko, na které když kliknete, tak se provede to, co je implementováno ve funkci.
+ *
+ */
+// Solution here
+const createButtonWithText = (buttonText) => {
+    const button = document.createElement('button');
+    button.innerText = buttonText;
+
+    return button;
+};
+
+const appendElementToSelected = (toBeAppended, appendTo) => {
+    const element = document.getElementById(appendTo);

```

Comment:
vzdycky pridavate paragraph do #result a tady ho vybirate pokazdem volani teto funkce, lepsi je provest vyber jednou a ulozit element do promenne

Reply:
Ok. Pro příště tedy. Chtěl jsem udělat funkci "univerzálnější," pokud by se např. používalo více různých divů.
Můžu tedy provést merge i takto?

Reply:
to uz je another story :)